ee/clickhouse/queries/clickhouse_funnel.py

```diff
@@ -60,16 +64,24 @@ def _exec_query(self) -> List[Tuple]:
         )
         self.params.update(prop_filter_params)
         steps = [self._build_steps_query(entity, index) for index, entity in enumerate(self._filter.entities)]
-        query = FUNNEL_SQL.format(
-            team_id=self._team.id,
-            steps=", ".join(steps),
-            filters=prop_filters.replace("uuid IN", "events.uuid IN", 1),
-            parsed_date_from=parsed_date_from,
-            parsed_date_to=parsed_date_to,
-            top_level_groupby="",
-            extra_select="",
-            extra_groupby="",
-            within_time="6048000000000000",
-            latest_distinct_id_sql=GET_LATEST_PERSON_DISTINCT_ID_SQL,
-        )
+
+        format_properties = {
+            "team_id": self._team.id,
+            "steps": ", ".join(steps),
+            "filters": prop_filters.replace("uuid IN", "events.uuid IN", 1),
+            "parsed_date_from": parsed_date_from,
+            "parsed_date_to": parsed_date_to,
+            "top_level_groupby": "",
+            "extra_select": "",
+            "extra_groupby": "",
+            "within_time": FunnelWindowDaysMixin.microseconds_from_days(self._filter.funnel_window_days),
+            "latest_distinct_id_sql": GET_LATEST_PERSON_DISTINCT_ID_SQL,
+            "offset": self._filter.offset,
+        }
+
+        if self._filter.offset == 0:
+            query = FUNNEL_SQL.format(**format_properties)
+        else:
+            query = FUNNEL_PERSONS_SQL.format(**format_properties)
```

Comment:
I think we should take this off the main funnel path and put funnel persons in a separate class or maybe a run method in this class like `get_people(...)`.

The frontend should be changed so that it retrieves people and the funnel data separately especially since we'll be paginating. The main reason I'm suggesting this is I'd prefer to avoid a default value of 0 or somehow if offset is passed from the frontend with a funnel payload it just starts returning people unexpectedly.

Reply:
I think I'll create a base class then have funnel and funnel_people extend it with a different SQL query.